cmd/advertisement-operator/main.go

```diff
@@ -163,10 +163,19 @@ func main() {
 		os.Exit(1)
 	}
 
+	newBroadcaster := &resourceRequestOperator.Broadcaster{}
+	componentStopper := make(chan struct{})
```

Comment:
You should be probably using contexts (WithCancel) instead of just channels if you need to stop a component

Reply:
This should no longer be needed if you use contexts